### PR TITLE
fix: env vars with fieldRef

### LINKED ISSUE #74: env var & fieldRef bug
Hi,

https://github.com/arttor/helmify/pull/72 released in https://github.com/arttor/helmify/releases/tag/v0.3.24 introduced a bug in the env vars using a `fieldRef`, such as

```yaml
        - name: OPERATOR_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
```

which gets converted to

```yaml
        - name: OPERATOR_NAMESPACE
          value: {{ .Values.controllerManager.manager.operatorNamespace }}
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
```

because we don't handle `fieldRef` and `resourceFieldRef` - see https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#envvarsource-v1-core

I'll open a PR tomorrow to fix it

### PR DESCRIPTION
**Pull request by vbehar**
add proper support for fieldRef and resourceFieldRef

fixes #74

### CODE REVIEW

**Review by arttor — CHANGES_REQUESTED** [review]
can you please regenerate samples for added env vars with:
```
cat test_data/k8s-operator-kustomize.output | go run ./cmd/helmify

**Comment by vbehar** [comment]
> can you please regenerate samples for added env vars with:
> 
> ```
> cat test_data/k8s-operator-kustomize.output | go run ./cmd/helmify 
> ```

done! thanks for the quick review

**Comment by arttor** [comment]
@vbehar please push sample changes, i can't find it in PR.

File: examples/operator/templates/deployment.yaml [hunk 1]
```diff
@@ -56,6 +56,20 @@ spec:
           value: {{ .Values.controllerManager.manager.var2 }}
         - name: VAR3_MY_ENV
           value: {{ .Values.controllerManager.manager.var3MyEnv }}
+        - name: VAR4
+          valueFrom:
+            configMapKeyRef:
+              key: VAR4
+              name: my-operator-configmap-vars
```
Comment (arttor) [inline examples/operator/templates/deployment.yaml:63]: configmap reference should also be templated if configmap is in the same chart. Please check out `secretKeyRef` sample

**Review by arttor — CHANGES_REQUESTED** [review]

File: examples/operator/templates/deployment.yaml (hunk 1 above)
Comment (vbehar) [reply examples/operator/templates/deployment.yaml]: done, sorry I missed that there was no configmap in the customize output, I just added one

**Review by arttor — APPROVED** [review]

**Pull request merged**